File: 747-min-cost-climbing-stairs/min-cost-climbing-stairs.java
class Solution {
    public int minCostClimbingStairs(int[] cost) {
        int prev1 = 0;
        int prev2 = 0;
        for(int i=0;i<cost.length;i++){
            int c1= Math.min(prev1 + cost[i],prev2 + cost[i]);
            prev2  = prev1;
            prev1 = c1;

        }
        return Math.min(prev1,prev2);

    }
}

File: 747-min-cost-climbing-stairs/min-cost-climbing-stairs.js
/**
 * @param {number[]} cost
 * @return {number}
 */
var minCostClimbingStairs = function(cost) {
    let ans=0;
    let arr=[];
    if(cost.length==2){
        if(cost[0]<cost[1]){
            return cost[0];
        }
        else {
            return cost[1]
        }
        
    } 

    let a=cost[0];
    arr.push(a);
    let b=cost[1];
    arr.push(b);
    for(let i=2;i<cost.length;i++){
        arr[i]=Math.min(arr[i-1],arr[i-2])+cost[i];
    
    }
    return Math.min(arr[cost.length-1],arr[cost.length-2]);
};